source/agora/common/crypto/ECC.d

```diff
@@ -90,11 +91,46 @@ public struct Scalar
         crypto_core_ed25519_scalar_reduce(this.data[].ptr, param[].ptr);
     }
 
-    /// Expose `toString`
-    public void toString (scope void delegate(const(char)[]) @safe sink)
-        const @safe
+    /// Describe print modes for `toString`
+    public enum PrintMode
+    {
+        /// Default value, print hashed version
+        Obfuscated,
+        /// Print original value
+        Clear,
+    }
+
+    /***************************************************************************
+
+        Print the scalar
+
+        By default, this function prints the hashed version of the scalar.
+        Changing the mode to `Clear` makes it print the non-hashed value.
+        This mode is intended for debugging.
+
+        Params:
+          sink = The sink to write the piecemeal string data to
+          mode = The `PrintMode` to use for printing the content.
+                 By default, the hashed value is printed.
+
+    ***************************************************************************/
+
+    public void toString (scope void delegate(const(char)[]) @safe sink,
+                          PrintMode mode = PrintMode.Obfuscated) const @safe
     {
-        this.data.toString(sink);
+        switch (mode)
+        {
+        case PrintMode.Obfuscated:
+            this.data[].hashFull().prettify().toString(sink);
+            break;
+
+        case PrintMode.Clear:
+            this.data.toString(sink);
+            break;
+
+        default:
+            assert(0);
+        }
```

Comment:
```suggestion
        final switch (mode)
        {
        case PrintMode.Obfuscated:
            this.data[].hashFull().prettify().toString(sink);
            break;

        case PrintMode.Clear:
            this.data.toString(sink);
            break;
        }
```

Final switch is a feature in D. If you have a missing `case` it will error at compile-time, rather than having the exception being thrown at runtime.